libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/backpress/DefaultOnBackPressedCallback.kt

```diff
@@ -13,70 +16,69 @@ import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
 import com.mapbox.navigation.ui.app.internal.navigation.NavigationStateAction
 import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
 import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
-import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
 /**
- * Key listener for the [NavigationView].
+ * Back pressed callback for the [NavigationView].
  *
  * On back pressed will update the navigation state.
  *
  * (FreeDrive) <- (DestinationPreview) <- (RoutePreview) <- (ActiveNavigation)
  *             <- (Arrival)
  */
 @ExperimentalPreviewMapboxNavigationAPI
-internal class OnKeyListenerComponent(
+internal class DefaultOnBackPressedCallback private constructor(
     private val store: Store,
-    private val view: View,
-    private val delegateOnKeyListener: View.OnKeyListener? = null
-) : UIComponent() {
+) : OnBackPressedCallback(false) {
 
-    override fun onAttached(mapboxNavigation: MapboxNavigation) {
-        super.onAttached(mapboxNavigation)
-
-        view.isFocusableInTouchMode = true
-        view.requestFocus()
-        view.setOnKeyListener { v, keyCode, event ->
-            if (delegateOnKeyListener?.onKey(v, keyCode, event) == true) {
-                return@setOnKeyListener true
-            }
-            if (keyCode == KeyEvent.KEYCODE_BACK && event.action == KeyEvent.ACTION_UP) {
-                handleBackPress()
-            } else {
-                false
-            }
-        }
-    }
-
-    private fun handleBackPress(): Boolean {
-        return when (store.state.value.navigation) {
+    override fun handleOnBackPressed() {
+        when (store.state.value.navigation) {
             NavigationState.FreeDrive -> {
-                false
+                error("The callback should be disabled in Free Drive state")
```

Comment:
Why error here? It seems like it should be a no-op.

Reply:
It is impossible to end up here, because the callback is disabled in free drive.

Reply:
Ok then why crash? Is there an actual error?

Reply:
If for some reason the callback becomes enabled in Free Drive (impossible with the current implementation), then nothing will happen when user presses the back button. Activity won't finish and Drop-In won't react in any way. This is clearly undesired behavior. But I can log a message instead of crashing.

Reply:
I would agree with logging instead of crashing.